eng/service.proj

```diff
@@ -62,6 +65,22 @@
   <ItemGroup Condition="'$(ProjectListOverrideFile)' != '' ">
     <Track1ProjectsToRemove Include="$(MSBuildThisFileDirectory)..\sdk\*\Microsoft.*.Management.*\**\*.csproj;$(MSBuildThisFileDirectory)..\sdk\*mgmt*\**\*.csproj;$(MSBuildThisFileDirectory)..\sdk\*\Microsoft.Azure.*\**\*.csproj" />
     <ProjectReference Remove="@(Track1ProjectsToRemove)" />
+    <TestProjects Include="..\sdk\$(ServiceDirectory)\$(Project)\tests\**\*.csproj" Exclude="@(MgmtExcludePaths)" />
+    <SamplesProjects Include="..\sdk\$(ServiceDirectory)\$(Project)\samples\**\*.csproj;..\sdk\$(ServiceDirectory)\samples\**\*.csproj" Exclude="@(MgmtExcludePaths)" />
+    <PerfProjects Include="..\sdk\$(ServiceDirectory)\$(Project)\perf\**\*.csproj" Exclude="@(MgmtExcludePaths)" />
+    <StressProjects Include="..\sdk\$(ServiceDirectory)\$(Project)\stress\**\*.csproj" Exclude="@(MgmtExcludePaths)"/>
+    <IntegrationTestProjects Include="..\sdk\$(ServiceDirectory)\$(Project)\integration\**\*.csproj" Exclude="@(MgmtExcludePaths)"/>
+    <SampleApplications Include="..\samples\**\*.csproj" Exclude="@(MgmtExcludePaths)"/>
+    <SharedProjects Include="..\sdk\$(ServiceDirectory)\$(Project)\shared\**\*.csproj" Exclude="@(MgmtExcludePaths)"/>
+    <SrcProjects Include="..\sdk\$(ServiceDirectory)\$(Project)\src\**\*.csproj" Exclude="@(TestProjects);@(SamplesProjects);@(PerfProjects);@(StressProjects)" />
+    <ProjectReference Remove="@(TestProjects)" Condition="'$(IncludeTests)' == 'false'" />
+    <ProjectReference Remove="@(SamplesProjects)" Condition="'$(IncludeSamples)' == 'false'" />
+    <ProjectReference Remove="@(PerfProjects)" Condition="'$(IncludePerf)' == 'false'" />
+    <ProjectReference Remove="@(StressProjects)" Condition="'$(IncludeStress)' == 'false'" />
+    <ProjectReference Remove="@(SampleApplications)" Condition="'$(IncludeSamplesApplications)' == 'false'"/>
+    <ProjectReference Remove="@(SrcProjects)" Condition="'$(IncludeSrc)' == 'false'" />
+    <ProjectReference Remove="@(IntegrationTestProjects)" Condition="'$(IncludeSrc)' == 'false'" />
+    <ProjectReference Remove="@(SharedProjects)" Condition="'$(IncludeSrc)' == 'false'" />
```

Comment:
This may break other use cases for ProjectListOverrideFile where the caller expected all of the projects in the file to be included without having to pass `/p:IncludeTests=true`, etc.

Reply:
see https://github.com/Azure/autorest.csharp/blob/440b1505086f9b0fc566745256743b596729e1c2/eng/UpdateAzureSdkCodes.ps1#L23

Reply:
So how am I supposed to make this work then? I have to have this exclusion logic to make the test invocations work.

Reply:
Good catch. I'm following up to see what the impact would be. It's possible I can just address this by forcing everything to true.

Reply:
Instead we went with a specific additional itemgroup that does the filtering. For most people using this thing the experience will be exactly the same as `main`.